prism/src/main/java/org/apache/falcon/resource/proxy/SchedulableEntityManagerProxy.java

```diff
@@ -378,6 +385,32 @@ protected APIResult doExecute(String colo) throws FalconException {
         return consolidateResult(results, APIResult.class);
     }
 
+    private void isEntityPartOfAnExtension(Entity entity) {
+        String tags = entity.getTags();
+        checkExtensionJobExist(tags);
+    }
+
+    private void checkExtensionJobExist(String tags) {
+        if (tags != null) {
+            String jobName = ExtensionManager.getJobNameFromTag(tags);
+            ExtensionMetaStore extensionMetaStore = ExtensionStore.getMetaStore();
+            if (jobName != null && extensionMetaStore.checkIfExtensionJobExists(jobName)) {
```

Comment:
During submit, checkIfExtensionJobExists might return false if that extension job is not submitted yet via the extension API. In such a case, the submit will go through..